### PR TITLE
[qtcontacts-sqlite] Isolate unit tests from real data

### PR DESCRIPTION
**Pull request by matthewvogt**
Also remove existing relevant test data before executing each unit test.

### CODE REVIEW

File: tests/tests.xml [hunk 1]
```diff
@@ -5,12 +5,15 @@
        <set name="unit-tests" feature="qtcontacts-sqlite-qt5">
            <description>Backend correctness automatic tests</description>
            <case manual="false" name="aggregation">
+               <step>rm -rf /home/nemo/.local/share/system/privileged/Contacts/qtcontacts-sqlite-test</step>
                <step>/opt/tests/qtcontacts-sqlite-qt5/tst_aggregation</step>
            </case>
            <case manual="false" name="contactmanager">
+               <step>rm -rf /home/nemo/.local/share/system/privileged/Contacts/qtcontacts-sqlite-test</step>
```
Comment (chriadam) [inline tests/tests.xml]: I don't like the hardcoding of the nemo user here, but no doubt we've done the same thing elsewhere, I guess.
It'd be nice if we could somehow parametrise the user home path, though.

**Comment by chriadam** [comment]
One comment, other than that LGTM.

File: tests/tests.xml (hunk 1 above)
Comment (matthewvogt) [inline tests/tests.xml]: It is possible, but it requires so much fiddling that I've regretted where I did the same in the past.  The path is so unlikely to change that making the file less readable (with replacement tokens) is not worth the cost.

**Comment by amtep** [comment]
LGTM

**Pull request merged**